pkg/controller/constrainttemplate/constrainttemplate_controller_test.go

```diff
@@ -316,12 +311,11 @@ func TestReconcile(t *testing.T) {
 		}
 	})
 
-	t.Run("Vap should not be created without label", func(t *testing.T) {
-		suffix := "VapShouldNotBeCreatedWithoutLabel"
+	t.Run("Vap should not be created without generateVAP", func(t *testing.T) {
```

Comment:
do we need to test for when a CT just has rego

Reply:
Added test for it.

Reply:
might be good to add another function similar to `makeReconcileConstraintTemplate` 
such that the rego is part of `code` instead of the top level
 https://github.com/JaydipGabani/gatekeeper/blob/dc49956f57c99af2a83a283c20a95a9f0df41f58/pkg/controller/constrainttemplate/constrainttemplate_controller_test.go#L84

Reply:
Added the test